src/calls/components/CallsView/Calendar.jsx

```diff
@@ -0,0 +1,241 @@
+import React, { Component } from 'react';
+import {
+    startOfMonth,
+    getDaysInMonth,
+    getDay,
+    getYear,
+    format,
+    addMonths,
+    subMonths,
+    addDays,
+    setMonth,
+    setYear
+} from 'date-fns';
+
+class Calendar extends Component {
+    state = {
+        date: new Date(),
+        months: [
+            'Jan',
+            'Feb',
+            'Mar',
+            'Apr',
+            'Maj',
+            'Jun',
+            'Jul',
+            'Avg',
+            'Sep',
+            'Okt',
+            'Nov',
+            'Dec'
+        ],
+        years: []
+    };
+
+    componentDidMount() {
+        this.props.handleChangeTableData(this.state.date);
+
+        if (this.state.years.length === 0) {
+            let years = [];
+            for (let y = 2000; y <= getYear(new Date()); y++) {
```

Comment:
You don't really need to keep an array of years as state. Treat dates as values, and use `date-fns` to add and subtract years.

Reply:
Ok, you are right about storing years in state.
I moved this part of code in render method, i hope it is ok?
But, I don't understand second sentence?

Reply:
Ok, I think I figured it out it.
Now I storing date objects in array which is not in component's state, but in component's render method.